visitor.go

```diff
@@ -61,11 +66,16 @@ func BFSWalk(d *DAG, visitor Visitor) {
 		queue.Enqueue(sv)
 	}
 
+	visited := make(map[string]bool, d.getOrder())
+
 	for !queue.Empty() {
 		v, _ := queue.Dequeue()
 		sv := v.(storableVertex)
 
-		visitor.Visit(sv)
+		if !visited[sv.WrappedID] {
+			visited[sv.WrappedID] = true
+			visitor.Visit(sv)
+		}
 
 		vertices, _ := d.getChildren(sv.WrappedID)
 		for _, id := range vertexIDs(vertices) {
```

Comment:
The function vertexIDs does two things. It makes the order of accessing the DAG unique, which is convenient for unit testing. Then it can make priority between sibling vertices, the smaller the id is the priority to visit, and the value of id can be used to implement a priority algorithm. reversedVertexIDs does the same.